staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java

```diff
@@ -132,6 +132,39 @@ public ResponseEntity<Void> deleteBatch(
         }
     }
 
+    @Override
+    public ResponseEntity<StatusResponse> getBatchStatus(
+            @ApiParam(value = "Identifies the tenant making the request.", required = true)
+            @RequestHeader(value = "X-TENANT-ID", required = true) String X_TENANT_ID,
+            @ApiParam(value = "Identifies the batch.", required = true)
+            @RequestParam("batchId") String batchId)
+    {
+        final StatusResponse status = new StatusResponse();
+        try {
+            final BatchDao.BatchStatus statusResponse = batchDao.getBatchStatus(new TenantId(X_TENANT_ID), new BatchId(batchId));
+            if (statusResponse == BatchDao.BatchStatus.COMPLETED) {
+                status.setMessage("Staging batch '" + batchId + "' is completed.");
+            } else if (statusResponse == BatchDao.BatchStatus.INPROGRESS) {
+                status.setMessage("Staging batch '" + batchId + "' is in progress.");
+            } else if (statusResponse == BatchDao.BatchStatus.ABANDONED) {
+                status.setMessage("Staging batch '" + batchId + "' is abandoned.");
```

Comment:
We could probably get away with this for abandoned batches but we could do much better.  How many abandoned batches do we have for this batch id?  For each of them, what was their start time (which we can easily get from the filename)?  Could we also report on the last modification time to know the time of abandonment?  I would also like if we reported on the abandoned batches even if there is a current batch in progress.